session.go

```diff
@@ -826,8 +859,8 @@ func (q *Query) attempt(keyspace string, end, start time.Time, iter *Iter, host
 			End:       end,
 			Rows:      iter.numRows,
 			Host:      host,
+			Metrics:   hostMetrics,
```

Comment:
I hope it's okay that the query observer will access the fields in `hostMetrics` while they could be changed – you're protecting write access above but that mutex is unlocked in line 852. Could that race? 🤔 😅